src/ssh_audit/ssh_audit.py

```diff
@@ -836,82 +839,129 @@ def process_commandline(out: OutputBuffer, args: List[str], usage_cb: Callable[.
+    parser.add_argument('-L', '--list-policies', action="store_true", dest='list_policies', default=None)
+    parser.add_argument('-M', '--make-policy', action="store", dest='make_policy', default=None)
+    parser.add_argument('-m', '--manual', action="store_true", dest='manual', default=None)
+    parser.add_argument('-P', '--policy', action="store", dest='policy', default=None)
+    parser.add_argument('-p', '--port', action="store", dest='port', default='22', type=int)
+    parser.add_argument('-T', '--targets', action="store", dest='targets', default=None)
+    parser.add_argument('-t', '--timeout', action="store", dest='timeout', default='5', type=int)
+    parser.add_argument('-v', '--verbose', action="store_true", dest='verbose', default=None)
+
+
+    # Add long options to the parser
+    parser.add_argument('--conn-rate-test', action="store", dest='conn_rate_test', default='0', type=int)
+    parser.add_argument('--dheat', action="store", dest='dheat', default='0', type=int)
+    parser.add_argument('--lookup', action="store", dest='lookup', default=None)
+    parser.add_argument('--no-colors', action="store_true", dest='no_colors', default=None)
+    parser.add_argument('--skip-rate-test', action="store_true", dest='skip_rate_test', default=None)
+    parser.add_argument('--threads', action="store", dest='threads', default='1', type=int)
+
+    try:
+        namespace = parser.parse_args()
+
+        if namespace.help is True:
             usage_cb(out)
-        elif o in ('-1', '--ssh1'):
-            aconf.ssh1 = True
-        elif o in ('-2', '--ssh2'):
-            aconf.ssh2 = True
-        elif o in ('-4', '--ipv4'):
-            aconf.ipv4 = True
-        elif o in ('-6', '--ipv6'):
-            aconf.ipv6 = True
-        elif o in ('-p', '--port'):
-            oport = a
-        elif o in ('-b', '--batch'):
+
+        aconf.host = namespace.host
+        host = namespace.host
+        port = namespace.port
+        aconf.ssh1 = namespace.ssh1
+        aconf.ssh2 = namespace.ssh2
+        aconf.ipv4 = namespace.ipv4
+        aconf.ipv6 = namespace.ipv6
+
+        aconf.json = namespace.json
+        if namespace.json_indent is True:
+            setattr(namespace, 'json', True)
+            aconf.json = namespace.json
+            aconf.json_print_indent = namespace.json_indent
+
+        if namespace.batch is True:
             aconf.batch = True
             aconf.verbose = True
-        elif o in ('-c', '--client-audit'):
-            aconf.client_audit = True
-        elif o in ('-j', '--json'):
-            if aconf.json:  # If specified twice, enable indent printing.
-                aconf.json_print_indent = True
-            else:
-                aconf.json = True
-        elif o in ('-v', '--verbose'):
+
+        aconf.client_audit = namespace.client_audit
+
+        ttime = namespace.timeout
+        if ttime != 5:
+            aconf.timeout = float(namespace.timeout)
```

Comment:
I suppose the `timeout` arg can be defined as a float in the `parser.add_argument()` function, above.

Reply:
I didn't think to try setting the add_argument as a float vs int.

I was using int's for other numbers and the original code converted the --timeout XX number to a float as seen above.

As you say this could be modified if desired.